### ISSUE TITLE
problem with date plugin

### ISSUE THREAD

**Issue by MohammadMahdiNiknam** [Kolcha/DigitalClock5#9]
i chose date and time in Persian but I doesn't show it properly. It is supposed to be from right to left but the text is left to right and this is not how it is supposed to be. for example, instead of:
چهارشنبه 25 شهریور 2024
it shows the texts like this:
2024 روی ره ش 25 ه ب ن ش را ه چ

**Comment by Kolcha** [comment]
yes, it will not work properly with right-to-left writing system, sorry. this is just because very simple and naive implementation. I'll see what I can do here to make it work properly.

you are the first user who contacted me about it :)

**Comment by Kolcha** [comment]
that's strange that it can handle RTL language correctly... older version has almost no text support at all. the new one was focused on text, but with very simple implementation, as I mentioned previously.

and yes, all previous versions since ~2016 are available on [SourceForge](https://sourceforge.net/projects/digitalclock4/files/)

**Comment by Kolcha** [comment]
just letting you know that the issue is not "forgotten", I'm working on improved skin rendering engine what will correctly handle RTL languages, here are the first results (I just hardcoded string you provided).

![Screenshot_20250105_122158](https://github.com/user-attachments/assets/cf638454-5df5-47c3-89fa-1005f395cdf0)

this engine uses similar tricks old clock used for text rendering (but there it was a set of hacks rather than "by design"), but still brings new customization options and uniform experience between image-based skins and font rendering.

new engine very likely will appear with version 5.1, not earlier than beginning of March (unfortunately, I don't have too much time to work on my own project).

**Comment by Kolcha** [comment]
addressed in 985c24c02ec43f04bf8349c5e5519b0b64714e8a with new engine, closing the issue
